src/main/java/org/terasology/manualLabor/systems/ShearingSystem.java

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.manualLabor.systems;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.core.Time;
+import org.terasology.engine.entitySystem.entity.EntityManager;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.event.ReceiveEvent;
+import org.terasology.engine.entitySystem.prefab.Prefab;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.RegisterSystem;
+import org.terasology.engine.logic.characters.events.AttackEvent;
+import org.terasology.engine.logic.delay.DelayManager;
+import org.terasology.engine.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.engine.registry.In;
+import org.terasology.engine.rendering.assets.material.Material;
+import org.terasology.engine.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.engine.rendering.logic.SkeletalMeshComponent;
+import org.terasology.gestalt.assets.ResourceUrn;
+import org.terasology.gestalt.assets.management.AssetManager;
+import org.terasology.manualLabor.components.ShearableComponent;
+
+import java.util.Optional;
+
+/**
+ * This system is responsible for handling the logic for shearing. A sheep shearing cycle consists of a shearing switching to the model for
+ * sheared state and switching back to the non sheared state after a certain amount of time.
+ */
+@RegisterSystem
+public class ShearingSystem extends BaseComponentSystem {
+    public static final String SHEARING_ITEM = "ManualLabor:CrudeShears";
+    public static final int HAIR_REGROWTH_TIME = 3 * 60 * 1000; // 3 minutes in ms
+    public static final String HAIR_REGROWTH_ACTION_ID = "ManualLabor:HairRegrowthAction";
+    public static final String SHEARED_SHEEP_MESH = "WildAnimals:shearedSheep";
+    public static final String SHEARED_SHEEP_MATERIAL = "WildAnimals:shearedSheepSkin";
+    public static final String SHEEP_MESH = "WildAnimals:sheep";
+    public static final String SHEEP_MATERIAL = "WildAnimals:sheepSkin";
+    private static final Logger logger = LoggerFactory.getLogger(ShearingSystem.class);
+
+    @In
+    protected Time time;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private DelayManager delayManager;
+
+    /**
+     * Executes changes occurring in event of shearing that is changing the sheared state if the entity is attacked by an appropriate shearing item.
+     * Also sets a periodic action responsible for triggering hair regrowth.
```

Comment:
```suggestion
     * Checks pre-conditions for shearing (e.g. the right tool) and initiates the shearing cycle.
```
As the shearing cycle is explained above, it's okay to just reference it here.